cordova-lib/src/cordova/plugin.js

```diff
@@ -206,10 +178,41 @@ module.exports = function plugin(command, targets, opts) {
                                     }
                                 }
 
-                                events.emit('verbose', 'Calling plugman.install on plugin "' + dir + '" for platform "' + platform + '" with options "' + JSON.stringify(options)  + '"');
-                                return plugman.raw.install(platform, platformRoot, path.basename(dir), pluginsDir, options);
+                                events.emit('verbose', 'Calling plugman.install on plugin "' + result.dest + '" for platform "' + platform + '" with options "' + JSON.stringify(options)  + '"');
+                                return plugman.raw.install(platform, platformRoot, path.basename(result.dest), pluginsDir, options);
                             });
-                        }, Q());
+                        }, Q()).then(function(){
+                            return Q(result);
+                        });
+                    })                    
+                    .then(function(result){
+                        metadata.save_fetch_metadata(pluginsDir, result.pinfo.id, { source: result.fetchJsonSource, variables: opts.cli_variables, is_top_level: true });
+                        return Q(result.dest);
+                    })
+                    .then(function(dir){
+                        // save to config.xml
+                        if(saveToConfigXmlOn(config_json,opts)){
```

Comment:
Usage: Function name `saveToConfigXmlOn` implies that it performs the saving, but your branch seems to do so.  Perhaps rename to `shouldSaveToConfigXml`.